lighthouse.go

```diff
@@ -995,74 +1044,96 @@ func (lhh *LightHouseHandler) handleHostQuery(n *NebulaMeta, vpnIp netip.Addr, a
 	}
 
 	if err != nil {
-		lhh.l.WithError(err).WithField("vpnIp", vpnIp).Error("Failed to marshal lighthouse host was queried for")
+		lhh.l.WithError(err).WithField("vpnAddrs", reqHostinfo.vpnAddrs).Error("Failed to marshal lighthouse host was queried for")
 		return
 	}
 
 	lhh.lh.metricTx(NebulaMeta_HostPunchNotification, 1)
-
-	//TODO: IPV6-WORK
-	binary.BigEndian.PutUint32(b[:], reqVpnIp)
-	sendTo := netip.AddrFrom4(b)
-	w.SendMessageToVpnIp(header.LightHouse, 0, sendTo, lhh.pb[:ln], lhh.nb, lhh.out[:0])
+	w.SendMessageToVpnIp(header.LightHouse, 0, reqVpnIp, lhh.pb[:ln], lhh.nb, lhh.out[:0])
 }
 
-func (lhh *LightHouseHandler) coalesceAnswers(c *cache, n *NebulaMeta) {
+func (lhh *LightHouseHandler) coalesceAnswers(v cert.Version, c *cache, n *NebulaMeta) {
 	if c.v4 != nil {
 		if c.v4.learned != nil {
-			n.Details.Ip4AndPorts = append(n.Details.Ip4AndPorts, c.v4.learned)
+			n.Details.V4AddrPorts = append(n.Details.V4AddrPorts, c.v4.learned)
 		}
 		if c.v4.reported != nil && len(c.v4.reported) > 0 {
-			n.Details.Ip4AndPorts = append(n.Details.Ip4AndPorts, c.v4.reported...)
+			n.Details.V4AddrPorts = append(n.Details.V4AddrPorts, c.v4.reported...)
 		}
 	}
 
 	if c.v6 != nil {
 		if c.v6.learned != nil {
-			n.Details.Ip6AndPorts = append(n.Details.Ip6AndPorts, c.v6.learned)
+			n.Details.V6AddrPorts = append(n.Details.V6AddrPorts, c.v6.learned)
 		}
 		if c.v6.reported != nil && len(c.v6.reported) > 0 {
-			n.Details.Ip6AndPorts = append(n.Details.Ip6AndPorts, c.v6.reported...)
+			n.Details.V6AddrPorts = append(n.Details.V6AddrPorts, c.v6.reported...)
 		}
 	}
 
 	if c.relay != nil {
-		//TODO: IPV6-WORK
-		relays := make([]uint32, len(c.relay.relay))
-		b := [4]byte{}
-		for i, _ := range relays {
-			b = c.relay.relay[i].As4()
-			relays[i] = binary.BigEndian.Uint32(b[:])
+		if v == cert.Version1 {
+			b := [4]byte{}
+			for _, r := range c.relay.relay {
+				if !r.Is4() {
+					continue
+				}
+
+				b = r.As4()
+				n.Details.OldRelayVpnAddrs = append(n.Details.OldRelayVpnAddrs, binary.BigEndian.Uint32(b[:]))
+			}
+
+		} else if v == cert.Version2 {
+			for _, r := range c.relay.relay {
+				n.Details.RelayVpnAddrs = append(n.Details.RelayVpnAddrs, netAddrToProtoAddr(r))
+			}
+
+		} else {
+			panic("unsupported version")
 		}
-		n.Details.RelayVpnIp = append(n.Details.RelayVpnIp, relays...)
 	}
 }
 
-func (lhh *LightHouseHandler) handleHostQueryReply(n *NebulaMeta, vpnIp netip.Addr) {
-	if !lhh.lh.IsLighthouseIP(vpnIp) {
+func (lhh *LightHouseHandler) handleHostQueryReply(n *NebulaMeta, reqHostinfo *HostInfo) {
+	//TODO: this is kind of dumb
```

Comment:
it is dumb, but it raises some good questions about multi-IP stuff. Are we going to require users to put _each_ IP for a lighthouse in their config file? If we do, does it make sense to group them?

```
lighthouse:
  hosts:
    - 10.0.0.1  # the old way
    - [fc00::2, 10.0.0.2]  # a new way
```
similar ideas apply wrt the static host map

Reply:
From a user perspective I would say no. I think it would be better to layer in learned lighthouse vpn addrs but as a stop gap I have included a `IsAnyLighthouseIP` (which the name is wrong ugh) that checks if any of the addrs are a lighthouse addr.